### PR TITLE
Add artifacts for Jupyterhub HA

### PR DESCRIPTION
**Pull request by maulikjs**
https://issues.redhat.com/browse/RHODS-1394

### CODE REVIEW

File: jupyterhub/jupyterhub/base/traefik-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        command:
+        - "/bin/bash"
+        - -c
+        args:
+        # this file has to exist before traefik starts
+        # so the watch can be set up properly
+        - touch /watched/rules.toml
+
+      containers:
+      - image: quay.io/internaldatahub/traefik:v1.7-build
```
Comment (crobby) [inline jupyterhub/jupyterhub/base/traefik-deployment.yaml:45]: @tmckayus I'm guessing this isn't the productized image here.  Will you need to add another magic sed at build-time to grab the right image?

File: jupyterhub/jupyterhub/base/traefik-deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            port: 8081
+          failureThreshold: 1
+          initialDelaySeconds: 10
+          periodSeconds: 10
+        resources:
+          limits:
+            cpu: 100m
+            memory: 50Mi
+
+      - image: quay.io/internaldatahub/configmap-puller:0.1
```
Comment (crobby) [inline jupyterhub/jupyterhub/base/traefik-deployment.yaml:83]: @tmckayus Same question here about magic sed.

**Review by crobby — COMMENTED** [review]
Assuming that we are ok using the /openshift/* images here instead of building our own.  Also had a couple of question/notes for Trevor about cpaas builds.

File: jupyterhub/jupyterhub/base/traefik-deployment.yaml (hunk 1 above)
Comment (tmckayus) [reply jupyterhub/jupyterhub/base/traefik-deployment.yaml:45]: Yes! good point.  We'll need the CPaaS build of traefik (quay.io/modh/odh-traefik-container)

File: jupyterhub/jupyterhub/base/traefik-deployment.yaml (hunk 2 above)
Comment (tmckayus) [reply jupyterhub/jupyterhub/base/traefik-deployment.yaml:83]: yes, same, we'll need the official build edited in

**Review by tmckayus — APPROVED** [review]

**Review by crobby — APPROVED** [review]

**Review by anishasthana — APPROVED** [review]

**Pull request merged**